pandas/core/indexes/base.py

```diff
@@ -5406,6 +5406,8 @@ def _get_indexer_strict(self, key, axis_name: str_t) -> tuple[Index, np.ndarray]
         self._raise_if_missing(keyarr, indexer, axis_name)
 
         keyarr = self.take(indexer)
+        if isinstance(key, Index):
+            keyarr.name = key.name
```

Comment:
can you add a comment leaving a breadcrumb back to here or wherever is appropriate?

Reply:
Good call, thanks. Added.